### PR TITLE
Save observational errors, subtype, duplicate factor from GEOS tests.

### PR DESCRIPTION
**Pull request by gmao-jjin3**
Save GsiInputObsError, GsiAdjustObsError,ObsSubType, Dupobs_Factor, and dup_kx_vector which contains ObsType (kx) numbers for duplicated observations.  These values are used in GMAO's JEDI/UFO configurations and tests.

### CODE REVIEW

**Review by danholdaway — APPROVED** [review]

File: src/gsi_ncdiag/gsi_ncdiag.py [hunk 1]
```diff
@@ -181,11 +181,13 @@
 
 gsi_add_vars_allsky = {
     'Observation_Type': 'ObsType',
+    'Observation_Subtype': 'ObsSubType',
     'Prep_Use_Flag': 'PreUseFlag',
     'Analysis_Use_Flag': 'GsiUseFlag',
     'Nonlinear_QC_Rel_Wgt': 'GsiQCWeight',
-    'Errinv_Adjust': 'GsiAdjustObsError',
```
Comment (CoryMartin-NOAA) [inline src/gsi_ncdiag/gsi_ncdiag.py:187]: Isn't this the 'default' GSI name though? Should we keep support for inverse error and input error?

**Comment by PatNichols** [comment]
@gmao-jjin3 Looks like your changes are not passing ctests. It probably just a problem with the input data. Here is the errors in the gsi_xxxx tests that are failing:
Platform:satwind Var:uv #Obs:71
Processed 71 Conventional obs processed to: testrun//satwind_obs_2018041500.nc4
DIFFER : NUMBER OF GROUPS : 13 ("/") <> 14 ("/")
DIFFER : Failed to find group /ObsSubType in file "testoutput/satwind_obs_2018041500.nc4"
DIFFER : Failed to find group /GsiAdjustObsError in file "testrun/satwind_obs_2018041500.nc4"
DIFFER : Failed to find group /GsiInputObsError in file "testrun/satwind_obs_2018041500.nc4"
Platform:sfc Var:tv #Obs:39
Processed 39 Conventional obs processed to: testrun//sfc_tv_obs_2018041500.nc4
Platform:sfc Var:tsen #Obs:1
Processed 1 Conventional obs processed to: testrun//sfc_tsen_obs_2018041500.nc4
DIFFER : NUMBER OF GROUPS : 12 ("/") <> 13 ("/")
DIFFER : Failed to find group /GsiAdjustObsError in file "testrun/sfc_tv_obs_2018041500.nc4"

Most likely just a case of the changing the input files....

**Comment by BenjaminRuston** [comment]
@gmao-jjin3 and @danholdaway is this still moving?  guess there are coding norm issues should be easy to fix, and it's having trouble finding `GSI` variables which it's trying to map to generic names?

**Comment by gmao-jjin3** [comment]
@PatNichols @BenjaminRuston All tests have passed now with current source data.  Thanks.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by BenjaminRuston — APPROVED** [review]
this looks reasonable let us know if there are issues

**Pull request merged**